src/host.c

```diff
@@ -626,6 +626,44 @@ cache_store (const char *host, struct address_list *al)
     }
 }
 
+bool
+resolve (const char *host, const char *ip)
+{
+  size_t len = strnlen(ip, 45);
+  bool validIpv4 = is_valid_ipv4_address(ip, ip + len);
+  bool validIpv6 = is_valid_ipv6_address(ip, ip + len);
+
+  if (validIpv4 == validIpv6) {
```

Comment:
You mean same address can't be valid ipv4 and ipv6 simultaneously? Not so obvious code - it breaks POLA (principle of least astonishement), I would say. At least insert an assertion to make the constraint clear:
```
assert((!validIpv4 && validIpv6) || validIpv4 != validIpv6);
```